src/components/marker-chart/Canvas.js

```diff
@@ -550,36 +551,90 @@ class MarkerChartCanvasImpl extends React.PureComponent<Props, State> {
       timelineTrackOrganization,
       viewport: { viewportLeft, viewportRight, viewportTop, containerWidth },
     } = this.props;
-    if (x < marginLeft - MARKER_DOT_RADIUS) {
+
+    // Note: we may want to increase this value to hit markers that are farther.
+    const dotRadius: CssPixels = MARKER_DOT_RADIUS * rowHeight;
+    if (x < marginLeft - dotRadius) {
       return null;
     }
+
     let markerIndex = null;
     let rowIndexOfLabel = null;
     const markerContainerWidth = containerWidth - marginLeft - marginRight;
 
     const rangeLength: Milliseconds = rangeEnd - rangeStart;
+
+    // Reminder: this is a value between 0 and 1, and represents a percentage of
+    // the full time range.
     const viewportLength: UnitIntervalOfProfileRange =
       viewportRight - viewportLeft;
-    const unitIntervalTime: UnitIntervalOfProfileRange =
+
+    // This is the x position in terms of unit interval (so, between 0 and 1).
+    const xInUnitInterval: UnitIntervalOfProfileRange =
       viewportLeft + viewportLength * ((x - marginLeft) / markerContainerWidth);
-    const time: Milliseconds = rangeStart + unitIntervalTime * rangeLength;
+
+    const dotRadiusInTime =
+      (dotRadius / markerContainerWidth) * viewportLength * rangeLength;
+
+    const xInTime: Milliseconds = rangeStart + xInUnitInterval * rangeLength;
     const rowIndex = Math.floor((y + viewportTop) / rowHeight);
-    const minDuration =
-      rangeLength *
-      viewportLength *
-      ((rowHeight * 2 * MARKER_DOT_RADIUS) / markerContainerWidth);
+
     const markerTiming = markerTimingAndBuckets[rowIndex];
 
-    if (!markerTiming || typeof markerTiming === 'string') {
+    if (
+      !markerTiming ||
+      typeof markerTiming === 'string' ||
+      !markerTiming.length
```

Comment:
I added this last case just to make sure. But actually I don't think this can happen, as our code will not add a new line if there's no marker to put there.